hera_cal/abscal_funcs.py

```diff
@@ -1055,56 +1063,83 @@ def interp2d_vis(model, model_lsts, model_freqs, data_lsts, data_freqs,
+
+    medfilt_flagged : type=bool, if True, run a median filter on flagged model pixels
 
-    fill_value : values to put for interpolation points outside training set
-        if None, values are extrapolated
+    medfilt_window : type=tuple, window for median filter across the (time, freq) axes.
 
-    zero_tol : for amplitudes lower than this tolerance, set real and imag components to zero
+    kx : type=int, type of spline to fit to freq axis of data. See scipy.interpolate.RectBivariateSpline
 
-    bounds_error : type=boolean, if True, raise ValueError when extrapolating. If False, extrapolate.
+    ky : type=int, type of spline to fit to time axis of data. See scipy.interpolate.RectBivariateSpline
 
-    flag_extrapolate : flag extrapolated data if True
+    s : type=int, smoothing factor for interpolation. Default is no smoothing. See scipy.interpolate.RectBivariateSpline
+
+    flag_extrapolate : type=bool, flag extrapolated data if True
 
     Output: (new_model, new_flags)
     -------
     new_model : interpolated model, type=dictionary
-    new_flags : flags associated with model, type=dictionary
+    new_flags : flags associated with interpolated model, type=dictionary
     """
     # make flags
     new_model = odict()
-    flags = odict()
+    new_flags = odict()
+
+    # get nearest neighbor points
+    freq_nn = np.array(map(lambda x: np.argmin(np.abs(model_freqs-x)), data_freqs))
+    time_nn = np.array(map(lambda x: np.argmin(np.abs(model_lsts-x)), data_lsts))
+    freq_nn, time_nn = np.meshgrid(freq_nn, time_nn)
 
     # loop over keys
     for i, k in enumerate(model.keys()):
-        # interpolate real and imag separately
+        # get model array
         m = model[k]
+
+        # get real and imag separately
         real = np.real(m)
         imag = np.imag(m)
 
-        # interpolate
-        interp_real = interpolate.interp2d(model_freqs, model_lsts, real,
-                                           kind=kind, fill_value=np.nan, bounds_error=bounds_error)(data_freqs, data_lsts)
-        interp_imag = interpolate.interp2d(model_freqs, model_lsts, imag,
-                                           kind=kind, fill_value=np.nan, bounds_error=bounds_error)(data_freqs, data_lsts)
         # set flags
-        f = np.zeros_like(interp_real, dtype=float)
-        if flag_extrapolate:
-            f[np.isnan(interp_real) + np.isnan(interp_imag)] = 1.0
-        interp_real[np.isnan(interp_real)] = fill_value
-        interp_imag[np.isnan(interp_imag)] = fill_value
+        if flags is not None:
+            f = flags[k][time_nn, freq_nn]
+            # check f is boolean type
+            if np.issubdtype(f.dtype, np.float):
+                f = ~(f.astype(np.bool))
+        else:
+            f = np.zeros_like(real, bool)
+         
+        # median filter flagged data if desired
+        if medfilt_flagged and flags is not None:
+            f_ext = int(medfilt_window[1]/2.)
```

Comment:
This has different behavior for even and odd medfilt_window sizes. Perhaps you want to clarify that in the documentation.